src/components/common/dashboard/Dashboard.vue

```diff
@@ -0,0 +1,324 @@
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                } as Operation,
+                {
+                    operationId: "FinishedMatriculation1",
+                    initiator: "MockUser4",
+                    initiatedTimestamp: "2011-10-05T14:48:00.000Z",
+                    lastModifiedTimestamp: "2020-10-05T16:48:00.000Z",
+                    transactionInfo: {
+                        contractName: UC4Identifier.CONTRACT_MATRICULATION,
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.FINISHED,
+                    reason: "",
+                    existingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                } as Operation,
+            ];
+
+            let mockedWatchedOps = [
+                {
+                    operationId: "FinishedMatriculation1",
+                    initiator: "MockUser4",
+                    initiatedTimestamp: "2011-10-05T14:48:00.000Z",
+                    lastModifiedTimestamp: "2020-10-05T16:48:00.000Z",
+                    transactionInfo: {
+                        contractName: UC4Identifier.CONTRACT_MATRICULATION,
+                        transactionName: UC4Identifier.TRANSACTION_ADD_MATRICULATION,
+                        parameters: "SS2020 Computer Science v3",
+                    },
+                    state: OperationStatus.FINISHED,
+                    reason: "",
+                    existingApprovals: {
+                        users: [],
+                        groups: ["Admin"],
+                    } as ApprovalList,
+                    missingApprovals: {
+                        users: [],
+                        groups: [],
+                    } as ApprovalList,
+                } as Operation,
+            ];
+
+            onBeforeMount(async () => {
+                await refresh();
+                await getEnrollmentId();
+                await getRole();
+            });
+
+            const isAdmin = computed(() => {
+                return role.value == Role.ADMIN;
+            });
+
+            async function refresh() {
+                busy.value++;
+                store.commit(MutationTypes.CLEAR_TREATED_OPERATIONS);
+                await getOperations();
+                busy.value--;
+            }
+
+            async function getRole() {
+                busy.value++;
+                role.value = (await store.getters.user).role;
+                busy.value--;
+            }
```

Comment:
correct me if I am wrong, but you only need role (in the html) to pass it down to other components, but thats what we have the store for, the store should be just as fast and simplifies the code here

Reply:
You are basically right. In this specific case though, we nee the role for displaying the button for opening the "AllOperations"-view. But the props were removed with https://github.com/upb-uc4/ui-web/pull/785/commits/af48219f5448569f497329fca5e0ffa5627e77db